velox/expression/SignatureBinder.cpp

```diff
@@ -122,90 +118,120 @@ bool SignatureBinder::tryBind() {
   return true;
 }
 
+bool SignatureBinder::tryAndSetIntegerParameter(
+    const std::string& parameterName,
+    int value) {
+  auto it = integerParameters_.find(parameterName);
+  // Return false if the parameter is not found.
+  if (it == integerParameters_.end()) {
+    return false;
+  }
+  // Return false if the parameter is found with a different value.
+  if (it->second.has_value() && it->second.value() != value) {
+    return false;
+  }
+  it->second = value;
+  return true;
+}
+
+bool SignatureBinder::tryBindIntegerParameters(
+    const std::vector<exec::TypeSignature>& parameters,
+    const TypePtr& actualType) {
+  // Only decimal types are supported.
+  if (!(actualType->isShortDecimal() || actualType->isLongDecimal())) {
+    return false;
+  }
+
+  VELOX_CHECK_EQ(parameters.size(), 2);
+  const auto& [precision, scale] = getDecimalPrecisionScale(*actualType);
+  return tryAndSetIntegerParameter(parameters[0].baseName(), precision) &&
+      tryAndSetIntegerParameter(parameters[1].baseName(), scale);
```

Comment:
This function name looks general, but the function body only handle Decimal types. Could we put these code for Decimal into a branch inside this function with the condition `if (actualType->isShortDecimal() || actualType->isLongDecimal())`?